editor/components/inserter/menu.js

```diff
@@ -28,40 +179,65 @@ class InserterMenu extends wp.element.Component {
 		return () => {
 			this.props.onInsertBlock( slug );
 			this.props.onSelect();
-			this.setState( { filterValue: '' } );
+			this.setState( {
+				filterValue: '',
+				focusedElementRef: null
+			} );
 		};
 	}
 
+	onKeyDown( keydown ) {
+		keydown.preventDefault();
+
+		if ( this.isNextKeydown( keydown ) ) {
+			this.focusNextFocusableRef( this );
+		}
+
+		if ( this.isPreviousKeydown( keydown ) ) {
+			this.focusPreviousFocusableRef( this );
+		}
+
+		if ( this.isEscapeKey( keydown ) ) {
+			this.props.closeMenu();
+		}
+	}
+
+	changeMenuSelection( refName ) {
+		this.setState( {
+			focusedElementRef: refName
+		} );
+
+		// Focus the DOM node.
+		this.nodes[ refName ].focus();
```

Comment:
Is there a reason why we're leveraging the "focus" to select a node? Is this for accessibility or something?

I guess I'd prefer to avoid the imperative approach and use a `className` or something.